### PR TITLE
[ci] [docs] add Oliver to CODEOWNERS

### PR DESCRIPTION
**Pull request by jameslamb**
Adds LightGBM's newest maintainer, @borchero , to CODEOWNERs.

Removes some unnecessary and out-of-date information from https://lightgbm.readthedocs.io/en/latest/FAQ.html.

### CODE REVIEW

File: .github/CODEOWNERS
Comment (jameslamb) [inline .github/CODEOWNERS:10]: @borchero this will automatically add you as a reviewer on every PR.

To be sure we set the right expectation...you do not need to review every PR! We require 1 approval from a maintainer to merge. In some situations, like more controversial changes or changes where particular maintainers have specialized knowledge, we sometimes ask for multiple approvals.

For example, for any non-trivial changes to distributed training, I'll probably ask for approvals from both @jmoralez (Dask knowledge) and one of @shiyu1994 and @guolinke (C++ and LightGBM internals knowledge).

File: docs/FAQ.rst
Comment (jameslamb) [inline docs/FAQ.rst:14]: These changes are included in this PR because I originally came to this page to add @borchero to the list of maintainers.

Explanation of the changes to these docs:

* none of this applies to only "Critical" issues, so removed language like that
* list of maintainers was seriously out-of-date
    - many of these people have not interacted with the repo in years
    - some of the notes about specialization were wrong (e.g. implying that I only work on R and Dask)
* example questions don't add much information above what we ask for in the issue templates (https://github.com/microsoft/LightGBM/tree/master/.github/ISSUE_TEMPLATE)

Comment (jameslamb) [reply docs/FAQ.rst:14]: I'm basically proposing changing this from "list of all maintainers" to "who you should `@` if you need timely help".

**Review by shiyu1994 — APPROVED** [review]

**Review by jmoralez — APPROVED** [review]

**Pull request merged**
